plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergReadVersionedTable.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.MaterializedResult;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.time.Instant;
+import java.util.Optional;
+
+import static io.trino.plugin.iceberg.IcebergQueryRunner.createIcebergQueryRunner;
+import static java.lang.String.format;
+
+public class TestIcebergReadVersionedTable
+        extends AbstractTestQueryFramework
+{
+    private long v1Unixtime;
+    private long v1SnapshotId;
+    private long v2Unixtime;
+    private long v2SnapshotId;
+
+    @Override
+    protected DistributedQueryRunner createQueryRunner()
+            throws Exception
+    {
+        return createIcebergQueryRunner();
+    }
+
+    @BeforeClass
+    public void setUp() throws InterruptedException
+    {
+        assertQuerySucceeds("CREATE SCHEMA test_schema");
+        assertQuerySucceeds("CREATE TABLE test_schema.test_iceberg_read_versioned_table (_string VARCHAR, _integer INTEGER)");
+        assertQuerySucceeds("INSERT INTO test_schema.test_iceberg_read_versioned_table VALUES ('a', 1)");
+        v1Unixtime = Instant.now().getEpochSecond() + 1; // go to the next second in case the operation finishes in the same second
+        v1SnapshotId = getLatestSnapshotId("test_schema", "test_iceberg_read_versioned_table").get();
+        Thread.sleep(2000);
```

Comment:
nit: since  `AS OF` can go up to milliseconds precision, I guess we can wait much less than 2 seconds here

Reply:
Please disregard this comment. I see that the logic in the tests makes checks with second precision timestamps. Sorry for not reading the tests well enough.

Reply:
Ok, sure.

Reply:
> Please refactor the test to sleep at most few millis.

bump